curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/inpaint.html

```diff
@@ -0,0 +1,118 @@
+<html>
+<head>
+<meta charset="utf-8">
+<title>CVdrawing</title>
+</head>
+<body>
+<p id="status">OpenCV.js is loading...</p>
+<canvas id="canvas" width="640" height="426">
+</canvas>
+<img id="src" src="bird.jpg" style="display:none"/>
+<input type="button" id="myButton" value="Test" disabled=true
+onclick="inpaint()"/>
+
+<script type="text/javascript">
+
+var checker = false;
+var src;
+var dst;
+var lastX = 0;
+var lastY = 0;
+var paint;
+var srcmat;
+var maskmat;
+var dstmat;
+var x;
+var y;
+
+window.onload = function()
+{
+context = document.getElementById("canvas").getContext("2d");
+img = document.getElementById("src");
+context.drawImage(img, 0, 0);
+};
+
+document.getElementById("canvas").onmousedown = function(e)
+{
+  paint = true;
+  x = e.pageX - this.offsetLeft;
+  y = e.pageY - this.offsetTop;
+  if (lastX == 0)
+  {
+      lastX = x;
+      lastY = y;
+  }
+};
+
+document.getElementById("canvas").onmousemove = function(e)
+{
+  if(paint)
+  {
+    cv.line(src, new cv.Point(lastX, lastY), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
+    cv.line(maskmat, new cv.Point(lastX, lastY), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
```

Comment:
Can you print `x, y, lastX, lastY` here? It seems to me that `x == lastX` and `y == lastY`.

Reply:
I print them - they are different.
They have same value in first point, then we start drawing.
For cv.line() we need previous point and next point.

Reply:
![Screenshot from 2019-08-15 11-25-20](https://user-images.githubusercontent.com/48130864/63087137-12824580-bf5a-11e9-8f7e-bdac12f7dff2.png)

Reply:
Please add delimiters between prints to see function scopes